test/e2e/metrics_test.go

```diff
@@ -0,0 +1,176 @@
+package e2e
+
+import (
+	"bufio"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/test/e2e/framework"
+)
+
+func setupMetricsEndpointTestCase(t *testing.T) (*framework.ClientSet, *operatorsv1.Console) {
+	client, _ := framework.StandardSetup(t)
+	routeForTest := tempRouteForTesting()
+	_, err := client.Routes.Routes(api.OpenShiftConsoleOperatorNamespace).Create(routeForTest)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("error: %s", err)
+	}
+	return client, nil
+}
+
+func cleanUpMetricsEndpointTestCase(t *testing.T, client *framework.ClientSet) {
+	routeForTest := tempRouteForTesting()
+	err := client.Routes.Routes(api.OpenShiftConsoleOperatorNamespace).Delete(routeForTest.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("error: %s", err)
+	}
+	framework.StandardCleanup(t, client)
+}
+
+// This test essentially does the following to verify the `console_url` is in the metrics endpoint:
+// 	curl --insecure -H "Authorization: Bearer $(oc whoami --show-token)" $(oc get route metrics --template 'https://{{.spec.host}}/metrics') | grep console_url
+// alternatively:
+// 	oc exec -it console-operator-<pod-id> /bin/bash
+// 	curl --insecure -H "Authorization: Bearer <token>  https://localhost:8443/metrics | grep console_url
+func TestMetricsEndpoint(t *testing.T) {
+	client, _ := setupMetricsEndpointTestCase(t)
+	defer cleanUpMetricsEndpointTestCase(t, client)
+
+	metricsURL := getMetricsURL(t, client)
+	fmt.Printf("fetching metrics url.... %v\n", metricsURL)
+
+	fmt.Printf("making request...\n")
+	respString := metricsRequest(t, metricsURL)
+
+	toFind := "console_url"
```

Comment:
Not sure if this shouldn't be a const after all the imports.

Also would make it a bit more clear that the string is just a metadata field, maybe!?:
```suggestion
const (
	metadataToFind = "console_url"
)
```

Reply:
Sounds good!